### PR TITLE
Admin ui 3

### PR DESCRIPTION
**Pull request by DominikSchmidtlein**
#### Problem
Desk layouts are currently just colored circles in squares in a rectangular area. This is hard to interpret and not very useful.

#### Solution
Allow the admin to upload a background. Since school already have floorplans they can upload them and overlay the desks over top of the floorplan. This will help the user understand where the squares and circles are physically located. In a future PR, the squares and circles should be little desk and grouping images respectively.

Contributes to #16 

 - [X] Tested locally?

### CODE REVIEW

File: app/controllers/groupings_controller.rb [hunk 1]
```diff
@@ -5,24 +5,34 @@ def show
...
-    @grouping.children.each do |area|
-      x, y = lay_out[area.type][area.id.to_s].values_at('x', 'y')
-      if area.coordinate
-        area.coordinate.x = x
-        area.coordinate.y = y
-        render_exception(area.coordinate) unless area.coordinate.save
-      else
-        area.create_coordinate(x: x, y: y)
-        render_exception(area) unless area.save
+    if params[:grouping]
```
Comment (erickadbay) [inline app/controllers/groupings_controller.rb]: `return render_exception(@grouping) if params[:grouping] && !@grouping.update(grouping_params)`

File: app/controllers/groupings_controller.rb [hunk 2]
```diff
@@ -5,24 +5,34 @@ def show
...
+      unless @grouping.update(grouping_params)
+        return render_exception(@grouping)
+      end
+    end
+    if params[:layout]
+      @grouping.children.each do |area|
+        x, y = coordinate(area)
+        if x.present? && y.present?
```
Comment (erickadbay) [inline app/controllers/groupings_controller.rb]: Do you need the `present?`? Otherwise you can just do `if x && y`

**Review by erickadbay — APPROVED** [review]
Beautiful work and tests ❤️ 

Great job!

File: app/controllers/groupings_controller.rb (hunk 2 above)
Comment (DominikSchmidtlein) [reply app/controllers/groupings_controller.rb]: No I think you're right.

**Comment by DominikSchmidtlein** [comment]
Thanks :nerd_face:

**Pull request merged**